src/main/java/org/isf/admission/rest/AdmissionController.java

```diff
@@ -0,0 +1,509 @@
+
+import io.swagger.annotations.Api;
+
+@RestController
+@Api(value = "/admissions", produces = "application/vnd.ohapi.app-v1+json")
+public class AdmissionController {
+
+	@Autowired
+	private AdmissionBrowserManager admissionManager;
+
+	@Autowired
+	private PatientBrowserManager patientManager;
+
+	@Autowired
+	private WardBrowserManager wardManager;
+
+	@Autowired
+	private DiseaseBrowserManager diseaseManager;
+
+	@Autowired
+	private OperationBrowserManager operationManager;
+
+	@Autowired
+	private PregnantTreatmentTypeBrowserManager pregTraitTypeManager;
+
+	@Autowired
+	private DeliveryTypeBrowserManager dlvrTypeManager;
+
+	@Autowired
+	private DeliveryResultTypeBrowserManager dlvrrestTypeManager;
+
+	private final Logger logger = LoggerFactory.getLogger(AdmissionController.class);
+
+	public AdmissionController(AdmissionBrowserManager admissionManager) {
+		this.admissionManager = admissionManager;
+	}
+
+	/**
+	 * Get {@link Admission} for the specified id
+	 * 
+	 * @param id
+	 * @return the {@link Admission} found or NO_CONTENT otherwise
+	 * @throws OHServiceException
+	 */
+	@GetMapping(value = "/admissions/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<AdmissionDTO> getAdmissions(@PathVariable int id) throws OHServiceException {
+		logger.info("Get admission by id:" + id);
+		Admission admission = admissionManager.getAdmission(id);
+		if (admission == null) {
+			return ResponseEntity.status(HttpStatus.NO_CONTENT).body(null);
+		}
+		System.out.println("admissiontype code" + admission.getAdmType().getCode());
+		return ResponseEntity.ok(getObjectMapper().map(admission, AdmissionDTO.class));
+	}
+
+	/**
+	 * Get the only one admission without Admission date for the specified patient
+	 * 
+	 * @param patientCode
+	 * @return found {@link Admission}, N0_CONTENT if there is no {@link Admission}
+	 *         found or message error
+	 * @throws OHServiceException
+	 */
+	@GetMapping(value = "/admissions/current", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<AdmissionDTO> getCurrentAdmission(@RequestParam("patientcode") Integer patientCode)
+			throws OHServiceException {
+		logger.info("Get admission by patient code:" + patientCode);
+		Patient patient = patientManager.getPatient(patientCode);
+		if (patient == null)
+			throw new OHAPIException(new OHExceptionMessage(null, "Patient not found!", OHSeverityLevel.ERROR),
+					HttpStatus.INTERNAL_SERVER_ERROR);
+		Admission admission = admissionManager.getCurrentAdmission(patient);
+		if (admission == null) {
+			return ResponseEntity.status(HttpStatus.NO_CONTENT).body(null);
+		}
+		return ResponseEntity.ok(getObjectMapper().map(admission, AdmissionDTO.class));
+	}
+
+	/**
+	 * 
+	 * @param searchTerms
+	 * @param admissionRange
+	 * @param dischargeRange
+	 * @return
+	 * @throws OHServiceException
+	 */
+	@GetMapping(value = "/admissions/admitted/patient", produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
I would suggest URL "/admissions/admittedPatients' for this one with pagination like in PatientController

Reply:
Hi @mwithi I updated the URL, and the core didn't consider pagination for this method.

Reply:
> Is there any reason why some methods from core have not been mapped?
> 
>     * getUsedWardBed(String wardId)
> 
>     * getNextYProg()
> 
> 
> probably there is plan to replace them by other business logic?
about this, I don't understand what this is all about!

Reply:
Ok, I've verified, and the pagination will be addressed by OP-161 👍

Reply:
> About this, I don't understand what this is all about!

`getUsedWardBed(String wardId)` is to know the beds occupation in a ward
`getNextYProgr()` is to query the database for next progressive number (within the Year) for an Admission

Reply:
Okay, the methods have been mapped.

Reply:
Thanks! I will review it in no time!

Reply:
You used the same mapping, I suggest:
@GetMapping(value = "/admissions/getNextProgressiveIdInYear"
@GetMapping(value = "/admissions/getBedsOccupationInWard"

Reply:
Okay, that's fixed.